src/FSharp.Build/WriteCodeFragment.fs

```diff
@@ -85,12 +90,47 @@ type WriteCodeFragment() as this =
                 else
                     Array.create (List.last orderedParametersWithIndex |> fst) "null"
 
-            List.iter (fun (index, value) -> orderedParametersArray.[index - 1] <- value) orderedParametersWithIndex
+            List.iter (fun (index, value) -> orderedParametersArray.[index - 1] <- value.Escaped) orderedParametersWithIndex
             // construct ordered parameter lists
             let combinedOrderedParameters = String.Join(", ", orderedParametersArray)
 
             let combinedNamedParameters =
-                String.Join(", ", List.map (fun (key, value) -> sprintf "%s = %s" key value) namedParameters)
+                // Define "_IsLiteral" suffix to match MSBuild behavior
+                let isLiteralSuffix = "_IsLiteral"
+                
+                // Process named parameters to handle IsLiteral suffix
+                let processedNamedParameters =
+                    // First identify all parameters with _IsLiteral suffix
+                    let isLiteralParams = 
+                        namedParameters
+                        |> List.choose (fun (key, _) -> 
+                            if key.EndsWith(isLiteralSuffix) && (namedParameters |> List.exists (fun (k, v) -> k = key && (v.Raw = "true" || v.Raw = "True"))) then
```

Comment:
Why is this quadratic list looping added? Just take the `value` in the lambda right above, where you have now used an `_` discard.